azurelinuxagent/common/osutil/mariner.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import os  # pylint: disable=W0611
+import re  # pylint: disable=W0611
+import pwd  # pylint: disable=W0611
+import shutil  # pylint: disable=W0611
+import socket  # pylint: disable=W0611
+import array  # pylint: disable=W0611
+import struct  # pylint: disable=W0611
+import fcntl  # pylint: disable=W0611
+import time  # pylint: disable=W0611
+import base64  # pylint: disable=W0611
+import errno
+import azurelinuxagent.common.conf as conf
+import azurelinuxagent.common.logger as logger  # pylint: disable=W0611
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil  # pylint: disable=W0611
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+from azurelinuxagent.common.exception import OSUtilError
+
+class MarinerOSUtil(DefaultOSUtil):
+    def __init__(self):
+        super(MarinerOSUtil, self).__init__()
+        self.agent_conf_file_path = '/etc/waagent.conf'
+        self.jit_enabled = True
+
+    def is_dhcp_enabled(self):
+        return True
+
+    def start_network(self) :
+        return shellutil.run("systemctl start systemd-networkd", chk_err=False)
```

Comment:
The `shellutil.run` command is being deprecated in favour of `azurelinuxagent.common.utils.shellutil.run_command` function. Please use that instead here and all the other functions below

Reply:
fixed